docs(alerting): Add two common examples in `Learn` section (#105325)
grafana/grafana: docs/sources/alerting/learn/examples/multi-dimensional-alerts.md

| CPU core | Active usage rate |
| :------- | :---------------- |
| cpu-0    | 95                |
| cpu-1    | 30                |
| cpu-2    | 85                |

This produces one series for each existing CPU.

When Grafana Alerting evaluates the query, it creates an individual alert instance for each returned series.

| Alert instance | Value |
| :------------- | :---- |
| {cpu="cpu-0"}  | 95    |
| {cpu="cpu-1"}  | 30    |
| {cpu="cpu-2"}  | 85    |

With a threshold condition like `$A > 80`, Grafana evaluates each instance separately and fires alerts only where the condition is met:

| Alert instance | Value | State  |
| :------------- | :---- | :----- |
| {cpu="cpu-0"}  | 95    | Firing |
| {cpu="cpu-1"}  | 30    | Normal |
| {cpu="cpu-2"}  | 85    | Firing |

Multi-dimensional alerts help you surface issues on individual components—problems that might be missed when alerting on aggregated data (like total CPU usage). Each alert instance includes distinct label data that identifies the component, making the alert more specific and actionable.

For example, Grafana might generate alerts like:

- High CPU usage on `cpu-0`

> - High CPU usage on `cpu-2`

Reviewer: It's not clear what "generate alerts" mean. Notifications?  Perhaps mention this example in the context of "description" annotation?

Author: addressed by https://github.com/grafana/grafana/pull/105325/commits/268ec1b1d402787237f6f19c69a3f5b72dabe1d0 - elaborate briefly the annotation example

{{< admonition type="note" >}}

> To avoid sending multiple notifications for similar alerts, Grafana Alerting [groups similar immediate alerts](ref:alert-grouping) into a single notification by default.

Reviewer: It's not clear what you mean when you say "immediate"

Author: addressed by https://github.com/grafana/grafana/pull/105325/commits/1751830f0feb013140707a37f43fb45005eca190

I decided removing the note about alert grouping to simplify this example which displays an annotation template with the recent changes.

{{< /admonition >}}

## Try it with TestData

You can quickly experiment with multi-dimensional alerts using the [**TestData** data source](ref:testdata-data-source), which can generate multiple random time series.

1. Add the **TestData** data source through the **Connections** menu.
1. Go to **Alerting** and create an alert rule
1. Select **TestData** as the data source.
1. Configure the TestData scenario

   1. Scenario: **Random Walk**
   1. Series count: 3
   1. Start value: 70, Max: 100
   1. Labels: `cpu=cpu-$seriesIndex`

   {{< figure src="/media/docs/alerting/testdata-random-series.png" max-width="750px" alt="Generating random time series data using the TestData data source" >}}

## Reduce time series data for comparison

The example returns three time series like shown above with values across the selected time range.

To alert on each series, you need to reduce the time series to a single value that the alert condition can evaluate and determine the alert instance state.

Grafana Alerting provides several ways to reduce time series data:

> - **Data source query functions**. In the earlier example, we used the Prometheus `sum` function to sum the rate results by `cpu,`producing a single value per CPU core.

Reviewer 2: ```suggestion
- **Data source query functions**. The earlier example used the Prometheus `sum` function to sum the rate results by `cpu,`producing a single value per CPU core.
```

- **Reduce expression**. In the query and condition section, Grafana provides the `Reduce` expression to aggregate time series data.
  - In **Default mode**, the **When** input selects a reducer (like `last`, `mean`, or `min`), and the threshold compares that reduced value.
  - In **Advanced mode**, you can add the [**Reduce** expression](ref:reduce-expression) (e.g., `last()`, `mean()`) before defining the threshold (alert condition).

> For demo purposes, we’ll use the **Advanced mode** with a **Reduce** expression:

Reviewer 2: ```suggestion
For demo purposes, this example uses the **Advanced mode** with a **Reduce** expression:
```
